File: app/src/main/java/com/blankj/androidutilcode/feature/core/bar/BarStatusActivity.java
package com.blankj.androidutilcode.feature.core.bar;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.blankj.androidutilcode.R;
import com.blankj.androidutilcode.base.BaseBackActivity;
import com.blankj.utilcode.util.BarUtils;
import com.blankj.utilcode.util.SpanUtils;

/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2016/10/13
 *     desc  : Activity 工具类 Demo
 * </pre>
 */
public class BarStatusActivity extends BaseBackActivity {

    private TextView tvAboutStatus;

    public static void start(Context context) {
        Intent starter = new Intent(context, BarStatusActivity.class);
        context.startActivity(starter);
    }

    @Override
    public void initData(Bundle bundle) {

    }

    @Override
    public int bindLayout() {
        return R.layout.activity_bar_status;
    }


    @Override
    public void initView(Bundle savedInstanceState, View view) {
        getToolBar().setTitle(getString(R.string.demo_bar));

        tvAboutStatus = findViewById(R.id.tv_about_status);
        findViewById(R.id.btn_show_status).setOnClickListener(this);
        findViewById(R.id.btn_hide_status).setOnClickListener(this);
        findViewById(R.id.btn_light_mode).setOnClickListener(this);
        findViewById(R.id.btn_dark_mode).setOnClickListener(this);
        updateAboutStatus();
    }

    @Override
    public void doBusiness() {

    }

    @Override
    public void onWidgetClick(View view) {
        switch (view.getId()) {
            case R.id.btn_show_status:
                BarUtils.setStatusBarVisibility(this, true);
                break;
            case R.id.btn_hide_status:
                BarUtils.setStatusBarVisibility(this, false);
                break;
            case R.id.btn_light_mode:
                BarUtils.setStatusBarLightMode(this, true);
                break;
            case R.id.btn_dark_mode:
                BarUtils.setStatusBarLightMode(this, false);
                break;
        }
        updateAboutStatus();
    }

    private void updateAboutStatus() {
        tvAboutStatus.setText(new SpanUtils()
                .appendLine("statusHeight: " + BarUtils.getStatusBarHeight())
                .append("isStatusVisible: " + BarUtils.isStatusBarVisible(this))
                .create());
    }
}

File: feature/utilcode/pkg/src/main/java/com/blankj/utilcode/pkg/feature/bar/status/BarStatusActivity.kt
package com.blankj.utilcode.pkg.feature.bar.status

import android.content.Context
import android.content.Intent
import com.blankj.common.activity.CommonActivity
import com.blankj.common.item.CommonItem
import com.blankj.common.item.CommonItemSwitch
import com.blankj.common.item.CommonItemTitle
import com.blankj.utilcode.pkg.R
import com.blankj.utilcode.util.BarUtils
import com.blankj.utilcode.util.CollectionUtils
import com.blankj.utilcode.util.Utils

/**
 * ```
 * author: Blankj
 * blog  : http://blankj.com
 * time  : 2016/10/13
 * desc  : demo about BarUtils
 * ```
 */
class BarStatusActivity : CommonActivity() {

    companion object {
        fun start(context: Context) {
            val starter = Intent(context, BarStatusActivity::class.java)
            context.startActivity(starter)
        }
    }

    override fun bindTitleRes(): Int {
        return R.string.demo_bar
    }

    override fun bindItems(): MutableList<CommonItem<*>> {
        return CollectionUtils.newArrayList(
                CommonItemTitle("getStatusBarHeight", BarUtils.getStatusBarHeight().toString()),
                CommonItemSwitch(
                        R.string.bar_status_visibility,
                        Utils.Func1 {
                            BarUtils.isStatusBarVisible(this)
                        },
                        Utils.Func1 {
                            BarUtils.setStatusBarVisibility(this, it)
                        }
                ),
                CommonItemSwitch(
                        R.string.bar_status_light_mode,
                        Utils.Func1 {
                            BarUtils.isStatusBarLightMode(this)
                        },
                        Utils.Func1 {
                            BarUtils.setStatusBarLightMode(this, it)
                        }
                )
        )
    }

    override fun onResume() {
        super.onResume()
        itemsView.updateItems(bindItems())
    }
}
